Orange/data/variable.py

```diff
@@ -243,12 +250,31 @@ def __init__(self, name="", compute_value=None):
         self.unknown_str = MISSING_VALUES
         self.source_variable = None
         self.attributes = {}
+        self.master = self
         if name and compute_value is None:
             if isinstance(self._all_vars, collections.defaultdict):
                 self._all_vars[name].append(self)
             else:
                 self._all_vars[name] = self
 
+    def make_proxy(self):
+        """
+        Copy the variable and set the master to `self.master` or to `self`.
+
+        :return: copy of self
+        :rtype: Variable
+        """
+        var = copy.copy(self)
```

Comment:
+1 for rename to derive.

Variable.derive is currently used in imputers, where we want a variable that looks the same, but with additional compute_value. For discretization, you need to define new DiscreteVariable with given values, so derive does not apply.

We can use copy.copy in variable.derive, if widgets are going to add new attributes  to variables.